Source/Core/VideoCommon/RenderState.h

```diff
@@ -93,42 +93,64 @@ union BlendingState
   u32 hex;
 };
 
-union SamplerState
+struct SamplerState
 {
-  using StorageType = u64;
+  void Generate(const BPMemory& bp, u32 index);
 
-  enum class Filter : StorageType
+  SamplerState& operator=(const SamplerState& rhs)
   {
-    Point,
-    Linear
-  };
+    tm0.hex = rhs.tm0.hex;
+    tm1.hex = rhs.tm1.hex;
+    return *this;
+  }
+
+  bool operator==(const SamplerState& rhs) const { return Hex() == rhs.Hex(); }
+  bool operator!=(const SamplerState& rhs) const { return Hex() != rhs.Hex(); }
+  bool operator<(const SamplerState& rhs) const { return Hex() < rhs.Hex(); }
 
-  enum class AddressMode : StorageType
```

Comment:
After thinking some more about this, from a generic high level, I do wish we hadn't bled the emulation state into our somewhat generic graphics api.  Obviously there is some overlap and this is relatively minor but I'm not sure this was necessary.  My assumption is the point of `Generate` was to make that conversion from emulation to graphics backend.  However, not suggesting you revert or modify this, just something I think we should avoid in the future.

To your point, adding a separate field might work but would make for a very klunky solution (if both are set, which do I choose?).  For this particular use case, wouldn't we have enough bits to add a new enum flag?  Again, no big deal, I will likely drop the feature (it's not critical) but thought it'd be worth discussing.

Reply:
There is an unused value in the enum (0, 1, and 2 correspond to valid values, while 3 is invalid), but hardware testing indicated that its behavior matches clamp (0).

Leaving the separate `AddressMode` enum would work, but then there would be an enum value that's not implemented in the actual shaders (since it doesn't exist on console), which feels weird to me.